core/auth/payment_metadata_test.go

```diff
@@ -0,0 +1,173 @@
+package auth_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/auth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEIP712Signer(t *testing.T) {
+	chainID := big.NewInt(17000)
+	verifyingContract := common.HexToAddress("0x1234000000000000000000000000000000000000")
+	signer := auth.NewEIP712Signer(chainID, verifyingContract)
+
+	privateKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	header := &core.PaymentMetadata{
+		BinIndex:          0,
+		CumulativePayment: 1000,
+		DataLength:        1024,
+		QuorumNumbers:     []uint8{1},
+	}
+
+	t.Run("SignPaymentMetadata", func(t *testing.T) {
+		signature, err := signer.SignPaymentMetadata(header, privateKey)
+		require.NoError(t, err)
+		assert.NotEmpty(t, signature)
```

Comment:
One thing you can do is use a contract script like this to generate the same signature on chain and compare. 
https://github.com/Layr-Labs/eigenda/blob/fab84e38d27c61d6c0cb66c77d25f6a9cfc0dc99/contracts/script/GenerateUnitTestHashes.s.sol#L13